docs/default.nix

```diff
@@ -37,11 +37,19 @@ let
       inherit lib;
 
       nixos-render-docs = prev.nixos-render-docs.overrideAttrs (old: {
+        propagatedBuildInputs = old.propagatedBuildInputs ++ [
+          (final.python3.pkgs.callPackage ./gfm-alerts-to-admonitions {
+            # We need to do this because nixos-render-docs does it internally, and it results in two
+            # different markdown-it-py derivations
+            markdown-it-py = final.python3.pkgs.markdown-it-py.overridePythonAttrs { doCheck = false; };
```

Comment:
I don't fully understand why we need to override `doCheck` here? Especially since we aren't overriding/overlaying `markdown-it-py` anywhere :face_with_diagonal_mouth:

Reply:
We don't, but nixos-render-docs [does](https://github.com/NixOS/nixpkgs/blob/b94f2597143d383aeed61a8f036b596d34a26dfc/pkgs/tools/nix/nixos-render-docs/default.nix#L7-L17).

I fiddled with a number of ways to pass our dependency, this one was the only one which did not raise a double-dependency error

Reply:
Is this because both our package and nixos-render-docs have a propagated dep on markdown-it-py? (sorry I'm still a bit confused)

Reply:
Yeah exactly, and they end up being _not_ the same package due to their override

Reply:
Maybe an alternative wording for the comments could be:

> Use the same override as `nixos-render-docs` does, to avoid "duplicate dependency" errors